apps/scratch/src/types/scratch-student-activities/delete/types.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  BaseStudentAppActivity,
+  BaseActivityRequest,
+  BaseStudentAction,
+  StudentAction,
+} from "../common";
+
+import {
+  DeletedBlockInfo,
+  DeletedBlockRecord,
+} from "../../../utilities/scratch-block";
+
+export interface StudentDeleteActivity extends BaseStudentAppActivity {
+  blockIdArray: DeletedBlockInfo | null;
```

Comment:
If something is typed as an array, I would suggest omitting the array suffix, no need to include the type in the name. On the other hand, plural may make sense: `blockIds`.

But effectively these are not just block ids so maybe `deletedBlocks`?